mmv1/products/migrationcenter/PreferenceSet.yaml

```diff
@@ -0,0 +1,197 @@
+              (Used Before) space required to store data before applying deduplication and
+              compression, in relation to the physical (Used After) space required after
+              applying deduplication and compression. Specifically, the ratio is the Used
+              Before space divided by the Used After space. For example, if the Used Before
+              space is 3 GB, but the physical Used After space is 1 GB, the deduplication
+              and compression ratio is 3x. Acceptable values are between 1.0 and 4.0. '
+          - !ruby/object:Api::Type::String
+            name: commitmentPlan
+            description: "Commitment plan to consider when calculating costs for virtual
+              machine insights and recommendations. If you are unsure which value to set,
+              a 3 year commitment plan is often a good value to start with. \n Possible
+              values:\n COMMITMENT_PLAN_UNSPECIFIED\nON_DEMAND\nCOMMITMENT_1_YEAR_MONTHLY_PAYMENTS\nCOMMITMENT_3_YEAR_MONTHLY_PAYMENTS\nCOMMITMENT_1_YEAR_UPFRONT_PAYMENT\nCOMMITMENT_3_YEAR_UPFRONT_PAYMENT"
+        name: vmwareEnginePreferences
+        description: 'The user preferences relating to Google Cloud VMware Engine target
+          platform. '
+      - !ruby/object:Api::Type::NestedObject
+        properties:
+          - !ruby/object:Api::Type::Double
+            name: cpuOvercommitRatio
+            description: 'CPU overcommit ratio. Acceptable values are between 1.0 and 2.0
+              inclusive. '
+          - !ruby/object:Api::Type::String
+            name: hostMaintenancePolicy
+            description: "Sole Tenancy nodes maintenance policy. \n Possible values:\n HOST_MAINTENANCE_POLICY_UNSPECIFIED\nHOST_MAINTENANCE_POLICY_DEFAULT\nHOST_MAINTENANCE_POLICY_RESTART_IN_PLACE\nHOST_MAINTENANCE_POLICY_MIGRATE_WITHIN_NODE_GROUP"
+          - !ruby/object:Api::Type::String
+            name: commitmentPlan
+            description: "Commitment plan to consider when calculating costs for virtual
+              machine insights and recommendations. If you are unsure which value to set,
+              a 3 year commitment plan is often a good value to start with. \n Possible
+              values:\n COMMITMENT_PLAN_UNSPECIFIED\nON_DEMAND\nCOMMITMENT_1_YEAR\nCOMMITMENT_3_YEAR"
+          - !ruby/object:Api::Type::Array
+            name: nodeTypes
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::String
+                  name: nodeName
+                  description: 'Name of the Sole Tenant node. Consult https://cloud.google.com/compute/docs/nodes/sole-tenant-nodes '
+            description: 'A list of sole tenant node types. An empty list means that all
+              possible node types will be considered. '
+        name: soleTenancyPreferences
+        description: 'Preferences concerning Sole Tenancy nodes and VMs. '
+    name: virtualMachinePreferences
```

Comment:
Unfortunately this isn't possible in ruby due to https://github.com/ruby/psych/issues/542

It is possible in old (1.9) versions of ruby, but we are way past that. Hopefully this will be enabled by a Go rewrite